From 05ff58606b084e897302b6cf93c4de8110ef6a17 Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@logicpd.com>
Date: Fri, 10 Jan 2020 10:08:34 -0600
Subject: [PATCH 43/43] arm64: dts: logicpd-imx8mm-kit: changes for Laird
 wireless module

Add subnodes and proper pin muxing for both WiFi and BT
---
 .../dts/freescale/lpd-imx8mm-baseboard.dtsi   |  1 -
 .../boot/dts/freescale/lpd-imx8mm-som.dtsi    | 55 ++++++++++++++++++-
 2 files changed, 54 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi b/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
index 95d97fa7f701..8bdcbf696b96 100644
--- a/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
@@ -33,7 +33,6 @@
 
 	modem_reset: modem-reset {
 		compatible = "gpio-reset";
-		reset-gpios = <&gpio2 6 GPIO_ACTIVE_LOW>;
 		reset-delay-us = <2000>;
 		reset-post-delay-ms = <40>;
 		#reset-cells = <0>;
diff --git a/arch/arm64/boot/dts/freescale/lpd-imx8mm-som.dtsi b/arch/arm64/boot/dts/freescale/lpd-imx8mm-som.dtsi
index ec6df119ab89..d0c9a1f83188 100644
--- a/arch/arm64/boot/dts/freescale/lpd-imx8mm-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/lpd-imx8mm-som.dtsi
@@ -1,4 +1,16 @@
 
+/ {
+	usdhc1_pwrseq: usdhc1_pwrseq {
+		compatible = "mmc-pwrseq-simple";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_usdhc1_gpio>;
+		reset-gpios = <&gpio2 10 GPIO_ACTIVE_LOW>; /* SD_1_RESET_B*/
+		clocks = <&osc_32k>;
+		clock-names = "ext_clock";
+		post-power-on-delay-ms = <80>;
+	};
+};
+
 &A53_0 {
 	arm-supply = <&buck2_reg>;
 };
@@ -226,8 +238,40 @@
 	assigned-clocks = <&clk IMX8MM_CLK_UART1>;
 	assigned-clock-parents = <&clk IMX8MM_SYS_PLL1_80M>;
 	fsl,uart-has-rtscts;
-	resets = <&modem_reset>;
 	status = "okay";
+
+	bluetooth {
+		compatible = "brcm,bcm43438-bt";
+		shutdown-gpios = <&gpio2 6 GPIO_ACTIVE_HIGH>;
+		host-wakeup-gpios = <&gpio2 8 GPIO_ACTIVE_HIGH>;
+		device-wakeup-gpios = <&gpio2 7 GPIO_ACTIVE_HIGH>;
+		clocks = <&osc_32k>;
+		clock-names = "extclk";
+	};
+};
+
+&usdhc1 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	bus-width = <4>;
+	non-removable;
+	cap-power-off-card;
+	pm-ignore-notify;
+	keep-power-in-suspend;
+	mmc-pwrseq = <&usdhc1_pwrseq>;
+	status = "okay";
+
+	brcmf: bcrmf@1 {
+		reg = <1>;
+		compatible = "brcm,bcm4329-fmac";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_wlan>;
+		interrupt-parent = <&gpio2>;
+		interrupts = <9 IRQ_TYPE_LEVEL_HIGH>;
+		interrupt-names = "host-wake";
+	};
 };
 
 &usdhc3 {
@@ -322,6 +366,9 @@
 				MX8MM_IOMUXC_UART3_RXD_UART1_DCE_CTS_B	0x140
 				MX8MM_IOMUXC_UART3_TXD_UART1_DCE_RTS_B	0x140
 				MX8MM_IOMUXC_SD1_DATA4_GPIO2_IO6	0x19
+				MX8MM_IOMUXC_SD1_DATA5_GPIO2_IO7	0x19
+				MX8MM_IOMUXC_SD1_DATA6_GPIO2_IO8	0x19
+				MX8MM_IOMUXC_GPIO1_IO00_ANAMIX_REF_CLK_32K	0x141
 			>;
 		};
 
@@ -417,6 +464,12 @@
 				MX8MM_IOMUXC_GPIO1_IO02_WDOG1_WDOG_B		0xc6
 			>;
 		};
+
+		pinctrl_wlan: wlangrp {
+			fsl,pins = <
+				MX8MM_IOMUXC_SD1_DATA7_GPIO2_IO9		0x111
+			>;
+		};
 	};
 };
 
-- 
2.17.1

